programs/psinode/main.cpp

```diff
@@ -2374,8 +2374,48 @@ void run(const std::string&              db_path,
          {
             if (!showedBootMsg)
             {
-               PSIBASE_LOG(node.chain().getLogger(), notice)
-                   << "Need genesis block; use 'psibase boot' to boot chain";
+               std::string xAdminSubdomain;
+               for (const auto& service : services)
+               {
+                  if (service.root.string().find("services/x-admin") != std::string::npos &&
+                      service.host.ends_with('.'))
+                  {
+                     xAdminSubdomain = service.host + host;
+                     break;
+                  }
+               }
+
+               if (!xAdminSubdomain.empty())
+               {
+                  std::string protocol;
+                  std::string port;
+                  for (const auto& listen : http_config->listen)
+                  {
+                     if (std::holds_alternative<psibase::http::tcp_listen_spec<true>>(listen))
```

Comment:
`std::get_if` can test and extract at once.

Reply:
Oh, nice. Done